### ISSUE TITLE
Ray install not optional

### ISSUE THREAD

**Issue by amm299** [adap/flower#799]
If you try to run the Quickstart-Pytorch example without installing Ray, the following error occurs:
```
  File "server.py", line 1, in <module>
    import flwr as fl
  File "/home/dux/.cache/pypoetry/virtualenvs/quickstart-pytorch-_Vro-tLH-py3.6/lib/python3.6/site-packages/flwr/__init__.py", line 18, in <module>
    from . import client, server, simulation
  File "/home/dux/.cache/pypoetry/virtualenvs/quickstart-pytorch-_Vro-tLH-py3.6/lib/python3.6/site-packages/flwr/simulation/__init__.py", line 18, in <module>
    from .app import start_simulation as start_simulation
  File "/home/dux/.cache/pypoetry/virtualenvs/quickstart-pytorch-_Vro-tLH-py3.6/lib/python3.6/site-packages/flwr/simulation/app.py", line 30, in <module>
    from flwr.simulation.ray_transport.ray_client_proxy import RayClientProxy
  File "/home/dux/.cache/pypoetry/virtualenvs/quickstart-pytorch-_Vro-tLH-py3.6/lib/python3.6/site-packages/flwr/simulation/ray_transport/ray_client_proxy.py", line 87, in <module>
    @ray.remote  # type: ignore
AttributeError: 'NoneType' object has no attribute 'remote'
```
It seems like the function definitions in ray_client_proxy.py need to be changed to handle the case Ray = None.

**Comment by tanertopal** [comment]
@amm299 thank you! We already have a PR which will fix that shortly (#806)

**Comment by danieljanes** [comment]
With #806 merged, Ray install is now optional (available in flwr-nightly and on branch `main`)
